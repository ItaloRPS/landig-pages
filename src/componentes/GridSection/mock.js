export default {
    background:false,
    title:'My grid',
    description:'Atque doloribud nobis laudantium esse ut,dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore',
    grid:[
        {
            title:'Teste 1',
            description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!'
            
        },  
        {
            title:'Teste 2',
            description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!'
            
        },
        {
            title:'Teste 3',
            description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!'
            
        }

    ]
}