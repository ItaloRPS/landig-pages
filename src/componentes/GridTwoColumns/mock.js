export default {
    title:'Grid Two Columns',
    text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Sit unde debitis vitae nesciunt praesentium ullam nulla saepe deleniti! Recusandae alias, praesentium numquam explicabo rerum esse vel? Explicabo, delectus. Perspiciatis, cupiditate.',
    srcImg: 'assets/imagens/javascript.svg'
}