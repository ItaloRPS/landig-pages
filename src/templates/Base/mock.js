import linksMock from '../../componentes/NavLinks/mock'
import gridMock from '../../componentes/GridSection/mock'
import {GridSection} from '../../componentes/GridSection'

export const mockBase = {
    children:(
        <>
            <GridSection {...gridMock}/>
            <GridSection {...gridMock} background/>
            <GridSection {...gridMock}/>
            <GridSection {...gridMock} background/>
        </>
    ),
        links:linksMock,
        logoData:{
            text:'Logo',
            link:'#'
        },
        footerHtml:'<p> Teste Footer </p>'
    
}
