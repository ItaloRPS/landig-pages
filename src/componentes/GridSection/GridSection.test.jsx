import { screen } from '@storybook/testing-library';
import {renderTheme} from '../../styles/render-theme'
import {GridSection} from '.'

import mock from './mock'

describe('<GridSection/>',()=>{
    test('shold render',()=>{
      const {container} =   renderTheme(<GridSection {...mock}/>);
        expect(container).toMatchSnapshot()
    });

    test('shold render without background',()=>{
        const {container} =   renderTheme(<GridSection {...mock} background={undefined}/>);
          expect(container).toMatchSnapshot()
      });
});

