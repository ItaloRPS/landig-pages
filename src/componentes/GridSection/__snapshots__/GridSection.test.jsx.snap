// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridSection/> shold render 1`] = `
.c2 {
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c7 {
  font-size: 3.2rem;
  text-transform: none;
}

.c4 {
  font-size: 2.4rem;
}

.c1 .c3 {
  margin-bottom: 6.4rem;
}

.c5 {
  counter-reset: grid-conunter;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(280px,1fr));
  gap: 3.2rem;
}

.c6 {
  position: relative;
  left: 5rem;
}

.c6::before {
  counter-increment: grid-conunter;
  content: counter(grid-conunter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
  font-weight: 800;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5rem;
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c5 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        My grid
      </h1>
      <div
        class="c3 c4"
      >
        Atque doloribud nobis laudantium esse ut,dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 1
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 2
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 3
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridSection/> shold render without background 1`] = `
.c2 {
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c7 {
  font-size: 3.2rem;
  text-transform: none;
}

.c4 {
  font-size: 2.4rem;
}

.c1 .c3 {
  margin-bottom: 6.4rem;
}

.c5 {
  counter-reset: grid-conunter;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(280px,1fr));
  gap: 3.2rem;
}

.c6 {
  position: relative;
  left: 5rem;
}

.c6::before {
  counter-increment: grid-conunter;
  content: counter(grid-conunter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
  font-weight: 800;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 5rem;
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c5 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        My grid
      </h1>
      <div
        class="c3 c4"
      >
        Atque doloribud nobis laudantium esse ut,dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 1
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 2
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
        <div
          class="c6"
        >
          <h3
            class="c7"
          >
            Teste 3
          </h3>
          <div
            class="c3 c4"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Id repudiandae non impedit asperiores sapiente! Consequatur accusamus mollitia error ratione harum dolores enim hic, doloremque labore inventore provident dolor dolorem at!
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
